bigfeta/bigfeta.py

```diff
@@ -172,6 +174,184 @@ def tilepair_weight(z1, z2, matrix_assembly):
     return tp_weight
 
 
+def _filter_to_products(contains_products, iterables_it):
+    """return list of values filtered to only those values which
+        are in the product of a list of iterables.
+
+    Parameters
+    ----------
+    contains_products : list
+        list to be filtered to values containing products
+    iterables_it : iterable of iterables
+        iterable of iterables as in inputs to itertools.product
+
+    Returns
+    -------
+    filtered_products : list
+        list of values in contains_products which are products of the iterables
+    """
+    return [p for p in contains_products
+            if all([p[i] in it for i, it in enumerate(iterables_it)])]
+
+
+def create_CSR_A_fromobjects(
```

Comment:
If we have this method, then it seems like we have some duplication of functionality with the class method `create_CSR_A`. I think the distinction now would be in one case, you're providing objects, and in another case, you're providing some data source from which to create the objects. So, it seems like the latter (the current one) could be rewritten to just extract from the data source and then call your new function, so the functionality is in only once place. It's been a while, maybe I am missing some intricacy.

Reply:
That's a fair point, and what I ended up doing with em_stitch.  The reason I didn't go that route is that `create_CSR_A` concurrently loads data and builds chunks in functions dependent on the database information.  It would be possible (and I think better in the long run) to make the module work like load data -> assemble -> solve, but that is a little bit more major of a change than just being able to run without connecting to a database.

Reply:
This is the point of #10, which also pointed out a bug here since I did not add a rough alignment test.  Once that passes tests it's likely that I will close this in favor of that PR.